templates/buildkite-elastic.yml

```diff
@@ -314,18 +320,28 @@ Resources:
                 start lifecycled
             02-install-buildkite:
               command: |
-                instance_id=\$(curl -sf http://169.254.169.254/latest/meta-data/instance-id)
-                instance_type=\$(curl -sf http://169.254.169.254/latest/meta-data/instance-type)
-                ami_id=\$(curl -sf http://169.254.169.254/latest/meta-data/ami-id)
-                docker_version=\$(docker --version | cut -f3 -d' ' | sed 's/,//')
-                meta_data_tags=\$(printf 'queue=%s,docker=%s,stack=%s,aws-buildkite-stack,aws:instance-id=%s,aws:instance-type=%s,aws:ami-id=%s' "$(BuildkiteQueue)" "\$docker_version" "$(AWS::StackName)" "\$instance_id" "\$instance_type" "\$ami_id")
-
-                sed -i -r "s/^(name)=.*/\\1=\"$(AWS::StackName)-\$instance_id\"/" /etc/buildkite-agent/buildkite-agent.cfg
-                sed -i -r "s/^(token)=.*/\\1=\"$(BuildkiteAgentToken)\"/" /etc/buildkite-agent/buildkite-agent.cfg
-                sed -i -r "s/^(# )?(meta-data-ec2-tags)=.*/\\2=true/" /etc/buildkite-agent/buildkite-agent.cfg
-                sed -i -r "s/^(# )?(meta-data)=.*/\\2=\"\$meta_data_tags\"/" /etc/buildkite-agent/buildkite-agent.cfg
-
-                service buildkite-agent start
+                INSTANCE_ID=\$(curl -sf http://169.254.169.254/latest/meta-data/instance-id)
+                DOCKER_VERSION=\$(docker --version | cut -f3 -d' ' | sed 's/,//')
+                BUILDKITE_AGENT_BINARY=/usr/bin/buildkite-agent-$(BuildkiteAgentRelease)
+                if [[ "$(BuildkiteAgentRelease)" == "stable" ]]; then
+                  BOOTSTRAP_SCRIPT=/etc/buildkite-agent/2-1-stable-bootstrap.sh
+                else
+                  BOOTSTRAP_SCRIPT="\$BUILDKITE_AGENT_BINARY bootstrap"
+                fi
+
+                cat << EOF > /etc/buildkite-agent
+                name="$(AWS::StackName)-\$INSTANCE_ID"
+                token="$(BuildkiteAgentToken)"
+                meta-data=\$(printf 'queue=%s,docker=%s,stack=%s,aws-buildkite-stack' "$(BuildkiteQueue)" "\$DOCKER_VERSION" "$(AWS::StackName)")
+                meta-data-ec2=true
+                meta-data-ec2-tags=true
```

Comment:
Perhaps this might have been a bad idea, we're already providing most of the useful tags. I feel like if we took this approach we should also be tagging the instances. Outside of the scope of this PR, but thoughts?